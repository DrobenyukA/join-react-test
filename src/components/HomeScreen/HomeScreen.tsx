import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

import routes from '../../constants/routes';

const ScreenLink = styled(NavLink)`
    display: inline-block;
    margin: 1rem 3rem;
    font-size: 2.75rem;
`;

const HomeScreenWrapper = styled.div`
    min-height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
`;

const HomeScreen = () => {
    return (
        <HomeScreenWrapper>
            <ScreenLink to={routes.APPLICANT}>Applicant screen</ScreenLink>
            <ScreenLink to={routes.CANDIDATES}>Candidate screen</ScreenLink>
        </HomeScreenWrapper>
    );
};

export default HomeScreen;
