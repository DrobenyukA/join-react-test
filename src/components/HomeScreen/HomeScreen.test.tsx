import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import HomeScreen from './HomeScreen';

describe('<HomeScreen />', () => {
    let wrapper: ShallowWrapper;

    beforeAll(() => {
        wrapper = shallow(<HomeScreen />);
    });

    it('Should have correct layout', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
