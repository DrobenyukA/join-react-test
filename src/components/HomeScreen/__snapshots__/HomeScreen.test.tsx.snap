// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> Should have correct layout 1`] = `
<styled.div>
  <Styled(NavLink)
    to="/applicant"
  >
    Applicant screen
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/candidates"
  >
    Candidate screen
  </Styled(NavLink)>
</styled.div>
`;
